sympy/matrices/expressions/tests/test_dotproduct.py

```diff
@@ -0,0 +1,8 @@
+from sympy.matrices import Matrix
+from sympy.matrices.expressions.dotproduct import DotProduct
+
+A = Matrix(3, 1, [1, 2, 3])
+B = Matrix(3, 1, [1, 3, 5])
+
+def test_docproduct():
+    assert DotProduct(A, B).doit() == 22
```

Comment:
And add tests for cases that should raise an exception, using `raises`. See https://github.com/sympy/sympy/blob/a57f3c43b9c86979ff7fe04d520bc404b94d859a/sympy/physics/vector/tests/test_output.py#L24 for an example of how it's used.